src/RedemptionPool.sol

```diff
@@ -98,74 +140,98 @@ contract RedemptionPool is Ownable {
-            revert RedemptionErrors.DeadlineExceeded();
-        }
+    function deposit(uint256 _amount) external onlyBeforeDeadline {
         // Transfers the GRO tokens from the sender to this contract
         GRO.safeTransferFrom(msg.sender, address(this), _amount);
         // Increases the balance of the sender by the amount
-        _userBalance[msg.sender] += _amount;
+        _userGROBalance[msg.sender] += _amount;
         // Increases the total deposited by the amount
         totalGRO += _amount;
-        // Emits the Deposit event
-        emit Deposit(msg.sender, _amount);
     }
 
     /// @notice withdraw deposited GRO tokens before the deadline
     /// @param _amount amount of GRO tokens to withdraw
-    function withdraw(uint256 _amount) external {
-        if (block.timestamp > DEADLINE) {
-            revert RedemptionErrors.DeadlineExceeded();
-        }
-        if (_userBalance[msg.sender] > _amount) {
-            revert RedemptionErrors.UserBalanceToSmall();
-        }
-        _userBalance[msg.sender] -= _amount;
+    function withdraw(uint256 _amount) external onlyBeforeDeadline {
+        if (_userGROBalance[msg.sender] < _amount)
+            revert RedemptionErrors.InsufficientBalance();
+
+        _userGROBalance[msg.sender] -= _amount;
         totalGRO -= _amount;
         GRO.safeTransfer(msg.sender, _amount);
         emit Withdraw(msg.sender, _amount);
     }
 
-    /// @notice Allow to withdraw cUSDC based on amount of GRO tokens deposited per user
-    function claim() external {
-        if (block.timestamp <= DEADLINE) {
-            revert RedemptionErrors.DeadlineExceeded();
-        }
-        if (_userBalance[msg.sender] <= 0) {
-            revert RedemptionErrors.NoUserBalance();
-        }
+    /**
+     * @notice Allow users to claim their share of USDC tokens based on the amount of GRO tokens they have deposited
+     * @dev Users must have a positive GRO token balance and a non-zero claim available to make a claim
+     * @dev The deadline for making GRO deposits must have passed
+     * @dev Redeems the user's cUSDC tokens for an equivalent amount of USDC tokens and transfers them to the user's address
+     * @dev Decreases the user's claims and contract accounting by the amount claimed
+     */
+    function claim(uint256 _amount) external onlyAfterDeadline {
+        // Get the amount of cUSDC tokens available for the user to claim
         uint256 userClaim = getSharesAvailable(msg.sender);
-        if (userClaim == 0) {
-            revert RedemptionErrors.NoUserClaim();
-        }
-        _userClaims[msg.sender] += userClaim;
-        IERC20(CUSDC).transfer(msg.sender, userClaim);
-        emit Claim(msg.sender, userClaim);
+
+        // Cap the user's claim to the available balance of cUSDC tokens
+        if (_amount > userClaim) _amount = userClaim;
+
+        // Revert if nothing to claim
+        if (_amount == 0) revert RedemptionErrors.InsufficientBalance();
```

Comment:
Can be checked at the very top to avoid spending gas if `_amount`  provided is 0

Reply:
At the current location it also reverts if userClaim is 0 - if you move it to top then it won't revert when userClaim is 0

Reply:
It does, but that requires additional SLOAD. Func parameters should be checked at the very top because they are very cheap to check if you need to revert